python-sdk/src/astro/databases/snowflake.py

```diff
@@ -475,35 +478,62 @@ def create_table_using_native_schema_autodetection(
             },
         )
 
+    @staticmethod
+    def get_cols_case(cols: tuple[str]):
+        result = []
+        for col in cols:
+            if col.isupper():
+                result.append(1)
+            elif col.islower():
+                result.append(0)
+            else:
+                return "mixed"
+
+        total = sum(result)
+        if total == 0:
+            return "lower"
+        elif total == len(cols):
+            return "upper"
+
+    @staticmethod
+    def use_quotes(cols: tuple[str]):
+        return SnowflakeDatabase.get_cols_case(cols) in ["upper", "mixed"]
```

Comment:
You could simplify this as below:

```suggestion
    @staticmethod
    def use_quotes(cols: tuple[str]):
        return any(col for col in cols if not col.islower())
```


Example tests:

```python
In [5]: use_quotes(["list", "sell"])
Out[5]: False

In [6]: use_quotes(["List", "sell"])
Out[6]: True

In [7]: use_quotes(["List", "Sell"])
Out[7]: True

In [8]: use_quotes(["List", "sEll"])
Out[8]: True

In [9]: use_quotes(["list", "sEll"])
Out[9]: True

In [10]: use_quotes(["LIST", "sEll"])
Out[10]: True

In [11]: use_quotes(["LIST", "sell"])
Out[11]: True

```

Reply:
Nice!